### PR TITLE
Create support image via autoyast with default patterns on SLE 12 SP5

### PR DESCRIPTION
**Pull request by hjluo**
- Description:
   * Create support image via autoyast with default patterns on SLE 12 SP5.

- Related MR: N/A
- Test Cases: [**520**](https://openqa.suse.de/admin/job_templates/520?)
- Related ticket: [**132650**](https://progress.opensuse.org/issues/132650)
- Needles: N/A
- Verification run: 
   - [**Autoyast_VRs**](https://openqa.suse.de/tests/overview?arch=&flavor=&machine=&test=sle_autoyast_support_image_gnome_12sp5&modules=&module_re=&distri=sle&version=12-SP5&build=20230820&groupid=520#)
   - Migration
        * [**x86_64**](https://openqa.suse.de/tests/12047335 )
        * [**aarch64**](https://openqa.suse.de/tests/12047339)
        * [**s390x**](https://openqa.suse.de/tests/12047332)
        * [**ppc64le**](https://openqa.suse.de/tests/12047337)

### CODE REVIEW

**Comment by lemon-suse** [comment]
I prefer to not select media installation, while to register scc to install with latest patches then de-register before publish qcow, and this will save time for patching in module combination jobs following.

**Comment by hjluo** [comment]
> I prefer to not select media installation, while to register scc to install with latest patches then de-register before publish qcow, and this will save time for patching in module combination jobs following.

Changed and updated the VRs please take a look again. thanks.

**Comment by chcao** [comment]
I think need to update the profile name by removing 'media'.

**Comment by hjluo** [comment]
> I think need to update the profile name by removing 'media'.

Done && thank you!

File: data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep [hunk 1]
```diff
@@ -0,0 +1,188 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (lemon-suse) [inline data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep:1]: Why still named as '.ep', I haven't seen the needed for it, please remove the extension name of 'ep' for all  the 4 files.

File: data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+<!DOCTYPE profile>
+<profile xmlns="http://www.suse.com/1.0/yast2ns" xmlns:config="http://www.suse.com/1.0/configns">
+    <suse_register>
+        <do_registration config:type="boolean">true</do_registration>
+        <reg_code><%= $get_var->('SCC_REGCODE') %></reg_code>
```
Comment (lemon-suse) [inline data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep:6]: Here should change to <reg_code>{{SCC_REGCODE}}</reg_code>

File: data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    <language>
+        <language>en_US</language>
+        <languages/>
+    </language>
+    <ntp-client>
+        <ntp_policy>auto</ntp_policy>
+    </ntp-client>
+    <software>
+        <products config:type="list">
+            <product><%= uc $get_var->('SLE_PRODUCT') %></product>
```
Comment (lemon-suse) [inline data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep:124]: change to {{SLE_PRODUCT}}

File: data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep (hunk 1 above)
Comment (hjluo) [reply data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep]: renamed.

File: data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep (hunk 2 above)
Comment (hjluo) [reply data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep]: changed.

File: data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep (hunk 3 above)
Comment (hjluo) [reply data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep]: changed.

File: data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml [hunk 4]
```diff
@@ -0,0 +1,188 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (lemon-suse) [inline data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml:1]: it should be default_patterns

Comment (lemon-suse) [inline data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml:1]: it should be default_patterns

**Comment by lemon-suse** [comment]
PUBLISH_HDD_1: 'autoyast-SLES-%VERSION%-%ARCH%-media-default-updated.qcow2'

s/media/GM

**Comment by lemon-suse** [comment]
Besides, need VRs of corresponding migration test to verify the qcow is ok.

File: data/yam/autoyast/support_images/sles12sp5_install_default_patterns_ppc64le.xml [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        <interface>
+          <bootproto>dhcp</bootproto>
+          <device>eth0</device>
+          <dhclient_set_default_route>yes</dhclient_set_default_route>
+          <startmode>auto</startmode>
+        </interface>
+      </interfaces>
+    </networking>
+    <firewall>
+      <enable_firewall config:type="boolean">true</enable_firewall>
```
Comment (rfan1) [inline data/yam/autoyast/support_images/sles12sp5_install_default_patterns_ppc64le.xml:36]: Is it required for `qemu` backend？

Comment (rfan1) [reply data/yam/autoyast/support_images/sles12sp5_install_default_patterns_ppc64le.xml:36]: IMO, we need to open the ssh port for s390x platform only

Comment (hjluo) [reply data/yam/autoyast/support_images/sles12sp5_install_default_patterns_ppc64le.xml]: keep it after we discuss this on slack. thanks

File: data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep (hunk 2 above)
Comment (hjluo) [reply data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep]: OK

File: data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml (hunk 4 above)
Comment (hjluo) [reply data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml]: done.

**Comment by hjluo** [comment]
> Besides, need VRs of corresponding migration test to verify the qcow is ok.

OK, waiting in the queue.

**Comment by hjluo** [comment]
@lemon-suse  @rfan1  now I've addressed all your comments and updated the latest VRs, please review it again, thanks.

**Comment by rfan1** [comment]
> @lemon-suse @rfan1 now I've addressed all your comments and updated the latest VRs, please review it again, thanks.

Seems you have defined `/home` as a subvolume on both x86-64 and ppc64le, however, based on the logs, /home is still shown as a separate partition. I am not sure why :)
Anyway, it didn't impact the test result, you can decide if you need to do some modifications [or just remove the `partitioning` part]

Others LGTM

**Comment by hjluo** [comment]
> > @lemon-suse @rfan1 now I've addressed all your comments and updated the latest VRs, please review it again, thanks.
> 
> Seems you have defined `/home` as a subvolume on both x86-64 and ppc64le, however, based on the logs, /home is still shown as a separate partition. I am not sure why :) Anyway, it didn't impact the test result, you can decide if you need to do some modifications [or just remove the `partitioning` part]
removed home subvolume. thanks
> Others LGTM

**Comment by lemon-suse** [comment]
LGTM, just the test case name of 'sle_autoyast_support_image_gnome_12sp5_def_patterns' should remove 'def_patterns', and qcow of 'autoyast-SLES-12-SP5-x86_64-GM-default-updated.qcow2' should remove 'default'. I will merge the PR, you still need one MR to update the qcow for corresponding migration test.

**Comment by hjluo** [comment]
> LGTM, just the test case name of 'sle_autoyast_support_image_gnome_12sp5_def_patterns' should remove 'def_patterns', and qcow of 'autoyast-SLES-12-SP5-x86_64-GM-default-updated.qcow2' should remove 'default'. I will merge the PR, you still need one MR to update the qcow for corresponding migration test.

OK, I've changed the test cases and it's setting for your comments.
[**job_group_520**](https://openqa.suse.de/admin/job_templates/520?)

**Review by rfan1 — APPROVED** [review]
LGTM, we are safe to merge once the VRs are done

**Review by lemon-suse — APPROVED** [review]

**Pull request merged**